pkg/operator/ceph/config/conditions.go

```diff
@@ -175,38 +187,20 @@ func ConditionInitialize(context *clusterd.Context, namespaceName types.Namespac
 	})
 }
 
-// conditionMapping maps the condition type to its status
-func conditionMapping(conditions []cephv1.Condition) {
-	for i := range conditions {
-		conditionType := conditions[i].Type
-		conditionMap[conditionType] = conditions[i].Status
-	}
-}
-
 // CheckConditionReady checks whether the cluster is Ready and returns the message for the Progressing ConditionType
 func CheckConditionReady(c *clusterd.Context, namespaceName types.NamespacedName) string {
-	cluster := &cephv1.CephCluster{}
-	err := c.Client.Get(context.TODO(), namespaceName, cluster)
+	cluster, err := getCluster(c, namespaceName)
 	if err != nil {
-		logger.Errorf("failed to get cluster %v", err)
-	}
-	if cluster.Status.Conditions != nil && len(conditionMap) == 0 {
-		conditionMapping(cluster.Status.Conditions)
-	}
-	if conditionMap[cephv1.ConditionReady] == v1.ConditionTrue {
-		return "Cluster is checking if updates are needed"
+		if cluster.Status.Conditions != nil {
+			conditionMap := conditionMapping(cluster.Status.Conditions)
+			if conditionMap[cephv1.ConditionReady] == v1.ConditionTrue {
+				return "Cluster is checking if updates are needed"
```

Comment:
In my testing I never see this "checking if updates are needed" on the `Progressing` condition. It's always "Cluster is creating".

Reply:
Yes, I was hesitant to change any existing behavior here.